app/services/idv/actions/redo_document_capture_action.rb

```diff
@@ -7,9 +7,14 @@ def self.analytics_submitted_event
 
       def call
         flow_session['redo_document_capture'] = true
-        unless flow_session[:skip_upload_step]
-          mark_step_incomplete(:link_sent)
+        if flow_session[:skip_upload_step]
+          redirect_to idv_document_capture_url
+        else
           mark_step_incomplete(:upload)
+
+          if !IdentityConfig.store.doc_auth_link_sent_controller_enabled
+            mark_step_incomplete(:link_sent)
+          end
```

Comment:
This change is smaller than it looks. Add a feature flag check to marking link_sent incomplete, and explicitly redirect to document capture if we're skipping upload, since the FSM final_url will be changing to link_sent in #8394.